src/Detector/DetectorLibrary/DotNetCore/DotnetCorePlatformDetector.cs

```diff
@@ -0,0 +1,146 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.Detector.Exceptions;
+
+namespace Microsoft.Oryx.Detector.DotNetCore
+{
+    internal class DotNetCorePlatformDetector : IPlatformDetector
+    {
+        private readonly DotNetCoreVersionProvider _versionProvider;
```

Comment:
These version providers tell the caller what versions are supported by Oryx. In case of detection, since we only 'detect', we need not use this.

Reply:
imo when we try to get version, it used the version providers to get version info, `detect `is very related to the version providers.